Fix BH performance: Remove unnecessary NOC_BRCST_EXCLUDE resets

msudumbrekar-TT opened the pull request:
Fixes #14828

This change removes unconditional NOC_BRCST_EXCLUDE register writes from general NOC write operations on Blackhole architecture. The register reset was causing 3-6% performance degradation for small transfers (< 2048 bytes).

Changes:
- Commented out NOC_BRCST_EXCLUDE writes in ncrisc_noc_fast_write()
- Commented out NOC_BRCST_EXCLUDE writes in ncrisc_noc_fast_write_loopback_src()
- Preserved the write in ncrisc_noc_fast_write_exclude_region() where it's needed

This aligns BH implementation with Quasar (tt-2xx) architecture where these writes are also removed.

Performance testing on BH hardware show improvements for several transfer sizes with no regressions.

msudumbrekar-TT committed:
Fix BH performance: Remove unnecessary NOC_BRCST_EXCLUDE resets

Fixes #14828

This change removes unconditional NOC_BRCST_EXCLUDE register writes from
general NOC write operations on Blackhole architecture. The register
reset was causing 3-6% performance degradation for small transfers
(< 2048 bytes).

Changes:
- Commented out NOC_BRCST_EXCLUDE writes in ncrisc_noc_fast_write()
- Commented out NOC_BRCST_EXCLUDE writes in ncrisc_noc_fast_write_loopback_src()
- Preserved the write in ncrisc_noc_fast_write_exclude_region() where it's needed

This aligns BH implementation with Quasar (tt-2xx) architecture where
these writes were already commented out.

Performance testing on BH hardware shows 1-2% improvements for several
transfer sizes with no regressions, confirming the fix is safe and
effective.

msudumbrekar-TT committed:
Address review: delete commented lines and remove ncrisc_noc_fast_write_exclude_region

abhullar-tt reviewed (approved):
could you also remove the same from quasar/noc_nonblocking_api.h?

msudumbrekar-TT committed:
Remove NOC_BRCST_EXCLUDE resets from quasar noc_nonblocking_api.h

- Remove commented NOC_BRCST_EXCLUDE lines from ncrisc_noc_fast_write and ncrisc_noc_fast_write_loopback_src
- Remove unused ncrisc_noc_fast_write_exclude_region function
